main.cpp

```diff
@@ -50,9 +50,47 @@ struct Wrapper
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
-
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")\n";
+    }
+
+    Type val;
+};
+
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")\n";
+    }
+
+    Point val;
+};
+
+template <typename T, typename ... Args>
+void variadicHelper(T&& firstArg, Args&& ... args)
+{
+    Wrapper<T> w (std::move(firstArg));
+    w.print();
+    
+    variadicHelper(std::forward<Args>(args) ... );
+}
+
+template <typename T>
+void variadicHelper(T&& firstArg)
+{
+    Wrapper<T> w (std::move(firstArg));
+    w.print();
+}

```

Comment:
There is another recursive solution.  it is possible for `Args ...` to be empty.
your task: remove your implementation for instruction `4)` and implement this alternate solution.
implement it **after** your existing `variadicHelper` function.
hint: what does the **recursive call** actually look like when called with an empty Variadic Parameter list?